react-app/src/features/welcome/index.tsx

```diff
@@ -18,15 +18,26 @@ export const loader = (queryClient: QueryClient) => {
     // Access specific parameters
     const errorDescription = queryParams.get("error_description");
     const error = queryParams.get("error");
+    // Check user query
+    const userQuery = queryClient.getQueryData<OneMacUser>(["user"]);
+    // Verify location point if logged out
+    const notLoggedOutPages =
+      window.location.pathname !== "/login" && window.location.pathname !== "/faq";
+
     if (errorDescription || error) {
       console.error("Authentication Error:", { errorDescription, error });
       return { error };
     }
-    if (!queryClient.getQueryData(["user"])) {
-      return await queryClient.fetchQuery({
+
+    if (!userQuery) {
+      const userFetch = await queryClient.fetchQuery({
         queryKey: ["user"],
         queryFn: () => getUser(),
       });
+      return userFetch?.user ? userFetch : notLoggedOutPages && redirect("/login");
+    }
```

Comment:
These are good changes. Although I also think we can move `await queryClient.fetchQuery` to the start of the function if it fetches from the cache when available.

I also think `['/login','faq'].includes(window.location.pathname) === false` could be easier to read

Reply:
Changed those and reworded nonLoggedIn as I didn't like it the more I stared at it. Thoughts?

Reply:
Still have to fix the tests now, but still.